### ISSUE TITLE
Investigate candidates for user analytics

### ISSUE THREAD

**Issue by liato** [liato/android-bankdroid#495]
To get a better understading of how our users use the app and what banks to prioritize we need to investigate which analytics service best suites our needs.

**Piwik** - http://piwik.org/
* Self hosted
* Open source
* Open API
* Extendable with plugins

**Google Analytics** - http://www.google.com/analytics/
* Proprietary
* Hosted by Google
* Open API

**Comment by JAGulin** [comment]
Indeed, this may be one of the most important changes. The "chase bank updates and depend on user reports" strategy seems to be a very costly process (even if I guess there is no alternative as long as the "banks" doesn't official support Bankdroid). Knowing what banks to focus on, and tracking errors automatically would allow some work improvements.

But, even if the users have already trusted that you'll treat their secrets properly, adding this kind of tracking should not be made silently. Be sure to ask for opt-in.

As for the choice of service you may want to consider also things like:
* architecture (strong support for app tracking or based on web tracking?)
* app bloat / size of library
* restrictions on data / uploads (rate limits? sampling or sending everything?)
* visualizations


Search web for "app analytics" or similar to find more alternatives. Or just go for Google Analytics and make sure that the data is not something worth "stealing" from you and the users.
